generator/ParamGenerator.cs

```diff
@@ -18,6 +18,7 @@ public void Execute(GeneratorExecutionContext context)
             var arborNodeType = context.Compilation.GetTypeByMetadataName("Arbor.Node");
             var arborBlackboardParameterType = context.Compilation.GetTypeByMetadataName("Arbor.BlackboardParameter`1");
             var listType = context.Compilation.GetTypeByMetadataName(typeof(List<>).FullName);
+            var arrayType = context.Compilation.GetTypeByMetadataName(typeof(System.Array).FullName);
```

Comment:
Would it make sense to actually replace both of those with the `IListType`  and check against that instead?

Reply:
This all turns out to be surprisingly finicky :V

The big problem is that I want(/need) to ensure proper typesafety, and for historical reasons, Array doesn't inherit from anything generic that indicates the proper type. Actually getting the proper Node[] type is also a royal pain; the code generation system is *not* the same as the runtime system, which is why I'm jumping through these get-type-by-name hoops.

It's possible there's a better way to do this, but if there is, I'm not sure what it is.